# Project: Profitable App Profile for the App Store and Google Play Markets

The company generates cash flow through ads shown in the application. Therefore, more users use apps; more ads mean more money.

The project aims to determine which kind of free mobile application generates more user engagement.

For our data analysis, we'll use the following datasets:

Apple Store datasets
https://dq-content.s3.amazonaws.com/350/AppleStore.csv

Google Store datasets
https://www.kaggle.com/datasets/lava18/google-play-store-apps

We are opening the Apple Store dataset saved on a CSV format file.

In [103]:
import csv

with open('AppleStore.csv') as file:
    csv_reader = csv.reader(file)
    apple_store_data = list(csv_reader)

An initial exploration of the Apple Store dataset

In [104]:
print(apple_store_data[2])

['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


We are opening the Google Play dataset saved on a CSV format file.

In [105]:
with open('googleplaystore.csv') as file:
    csv_reader = csv.reader(file)
    google_play_data = list(csv_reader)

An initial exploration of the Google Play dataset

In [106]:
print(google_play_data[1])

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


We are defining a function to explore data on a dataset.

In [107]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

We are exploring the first 4 rows of the Apple Store dataset.

In [108]:
print(explore_data(apple_store_data, 1, 5))

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


None


We are exploring the first 4 rows of the Google Play dataset.

In [109]:
print(explore_data(google_play_data, 1, 5))

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


None


We explore the first four rows of the Apple Store dataset and return the total number of rows and columns.

In [110]:
print(explore_data(apple_store_data, 1, 5, True))

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7198
Number of columns: 16
None


We explore the first four rows of the Google Play dataset and return the total number of rows and columns.

In [111]:
print(explore_data(google_play_data, 1, 5, True))

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10842
Number of columns: 13
None



We are exploring the first role of the Apple Store dataset, which is the header of the CSV file.

In [112]:
print(apple_store_data[0])

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


We are exploring the first role of the Google Play dataset, which is the header of the CSV file.

In [113]:
print(google_play_data[0])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


Row 10473 contains an error. If you check carefully, the row doesn't have value for the "Category" column.

In [114]:
print(google_play_data[10473])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


## Working with Google Play dataset
#### Cleaning process with incorrect data

We are deleting row 10473 from the Google Play data because it contains an error that will affect our analysis process.


In [115]:
print(len(google_play_data))
del google_play_data[10473]
print(len(google_play_data))

10842
10841


#### Cleaning process with duplicated data (Google Play dataset)

Finding duplicated rows in the Google Play Data set.

The code below shows how many rows (duplicated) for the Instagram app are inside the Google Play dataset:

In [116]:
for app in google_play_data[1:]:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


The code below shows how many rows (duplicated) for the Facebook app are inside the Google Play dataset:

In [117]:
for app in google_play_data[1:]:
    name = app[0]
    if name == 'Facebook':
        print(app)

['Facebook', 'SOCIAL', '4.1', '78158306', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']
['Facebook', 'SOCIAL', '4.1', '78128208', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']



The code bellow find the total amount of duplicated rows in the Google Play dataset

In [118]:
duplicated_apps = list()
unique_apps = list()

for row in google_play_data[1:]:
    if row[0] not in unique_apps:
        unique_apps.append(row[0])
    else:
        duplicated_apps.append(row[0])

In [119]:
print("There {quantity} duplicated rows in the Google Play dataset".format(quantity=len(duplicated_apps)))

There 1181 duplicated rows in the Google Play dataset


To clean the Google Play dataset from duplicated rows, we'll remove all duplicated rows except the one with the higher number of reviews (column with index 3). For example, we'll keep the row with 66577446 reviews for the Instagram case.

In [120]:
for app in google_play_data[1:]:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


The code below creates a dictionary containing the app's name and the number of reviews. The dictionary stores the row with the higher number of reviews.

In [121]:
reviews_max = dict()

for row in google_play_data[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if (name in reviews_max) and (reviews_max[name] < n_reviews):
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
    
    

The expected length of the dictionary is 9,659 items.

In [122]:
print("The dictionary has {keys} items".format(keys=len(reviews_max)))

The dictionary has 9659 items


The following code will create a new Google Play dataset (a list of lists) based on data stored in the dictionary called 'reviews_max'. The dictionary contains a unique application name with a higher number of reviews.

In [123]:
android_clean = list()
already_added = list()

for row in google_play_data[1:]:
    name = row[0]
    n_reviews = float(row[3])    
    if (n_reviews == reviews_max[name]) and (name not in already_added): 
        android_clean.append(row)
        already_added.append(name)

The expected length of the new list (android_clean) is 9,659 elements.

In [124]:
print("The lenght of the android_clean dictionary is {value}".format(value=len(android_clean)))

The lenght of the android_clean dictionary is 9659


The following function checks whether a character has a value higher than 127 (ASCII). If so, the word is considered a non-English word.

In [125]:
def is_english_word(input_string):
    for c in input_string:
        if ord(c) > 127:
            return False
    return True        

Testing #1 of the function is_english_word:

In [126]:
print(is_english_word('Instagram'))

True


Testing #2 of the function is_english_word:

In [127]:
print(is_english_word('爱奇艺PPS -《欢乐颂2》电视剧热播'))

False


Testing #3 of the function is_english_word:

In [128]:
print(is_english_word('Docs To Go™ Free Office Suite'))

False


Testing #4 of the function is_english_word:

In [129]:
print(is_english_word('Instachat 😜'))

False


We modify the previous function. This time if a string contains more than three characters whose ASCII values are higher than 127, the word is considered non-English.

In [130]:
def is_english_word_v2(input_string):
    counter = 0
    for c in input_string:
        if ord(c) > 127:
            counter += 1
    if counter > 3:
        return False
    return True
            
            

Testing #1 of the function is_english_word:

In [131]:
print(is_english_word_v2('Docs To Go™ Free Office Suite'))

True


Testing #2 of the function is_english_word:

In [132]:
print(is_english_word_v2('Instachat 😜'))

True


Testing #3 of the function is_english_word:

In [133]:
print(is_english_word_v2('爱奇艺PPS -《欢乐颂2》电视剧热播'))

False


The following code removes applications with non-English names from the androd_clean dataset.

In [134]:
android_clean_non_english_apps = list()
android_clean_english_apps = list()
for row in android_clean:
    if is_english_word_v2(row[0]) == False:
        android_clean_non_english_apps.append(row)
    else:
        android_clean_english_apps.append(row)
    

In [135]:
print("The total number of non-English apps is {quantity}".format(quantity=len(android_clean_non_english_apps)))

The total number of non-English apps is 45


In [136]:
print("The total number of English apps is {quantity}".format(quantity=len(android_clean_english_apps)))

The total number of English apps is 9614


The code below isolates free apps to a separate list in Google Play.

In [137]:
android_clean_english_apps_free = list()
for row in android_clean_english_apps:
    if (row[6].lower()) == 'free':
        android_clean_english_apps_free.append(row)
        
    

In [138]:
print("There are {quantity} free apps in the Google Play dataset".format(quantity=len(android_clean_english_apps_free)))

There are 8863 free apps in the Google Play dataset


### Final Google Play dataset

In [139]:
google_play_data_final = android_clean_english_apps_free
google_header = google_play_data[0]
google_play_data_final.insert(0, google_header)

In [140]:
print("The total number of elements on the final Google Play dataset is {quantity}".format(quantity=len(google_play_data_final)))

The total number of elements on the final Google Play dataset is 8864


### Working with Apple Store dataset
#### Cleaning process with incorrect data

The code below isolates non-English apps to a separate list from the Apple Store dataset.

In [141]:
apple_store_non_english_apps = list()
apple_store_english_apps = list()
for row in apple_store_data[1:]:
    if is_english_word_v2(row[1]) == False:
        apple_store_non_english_apps.append(row)
    else:
        apple_store_english_apps.append(row)

In [142]:
print("The total number of English apps is {quantity}".format(quantity=len(apple_store_english_apps)))

The total number of English apps is 6183


In [143]:
print("The total number of English apps is {quantity}".format(quantity=len(apple_store_non_english_apps)))

The total number of English apps is 1014


The code below isolates free apps to a separate list in Apple Store.

In [144]:
apple_store_english_apps_free = list()
apple_store_english_apps_non_free = list()

for row in apple_store_english_apps:
    if float(row[4]) == 0.0 or float(row[4]) == 0:
        apple_store_english_apps_free.append(row)
    else:
        apple_store_english_apps_non_free.append(row)

In [145]:
print("The total number of English apps is {quantity}".format(quantity=len(apple_store_english_apps_free)))

The total number of English apps is 3222


In [146]:
print("The total number of English apps is {quantity}".format(quantity=len(apple_store_english_apps_non_free)))

The total number of English apps is 2961


### Final Apple Store dataset

In [147]:
apple_store_final = apple_store_english_apps_free
apple_store_header = apple_store_data[0]
apple_store_final.insert(0, apple_store_header)

In [148]:
print("The total number of elements on the final Google Play dataset is {quantity}".format(quantity=len(apple_store_final)))

The total number of elements on the final Google Play dataset is 3223


In [149]:
print(apple_store_final[0])

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [150]:
print(google_play_data_final[0])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


### Data Analysis part #1

The code below represents functions that take two input values: a) a dataset; b) an index. Then, the function returns a frequency table whose values are represented in percentages.

In [151]:
def freq_table(dataset, index):
    freq_table = dict()    
    for row in dataset[1:]:        
        if row[index] in freq_table:
            freq_table[row[index]] += 1
        else:
            freq_table[row[index]] = 1
    
    for item in freq_table:
        freq_table[item] = ((freq_table[item] / len(dataset))) * 100        
    return freq_table

        

We create a frequency table from the Google Play dataset based on the column "Genres" (index 9):

In [152]:
googe_freq_table = freq_table(google_play_data_final, 9)

The code below represents functions that take two input values: a) a dataset; b) an index. Then, it returns a frequency sorted in descending order.

In [153]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

Below, we can see the most popular apps in Google Play by Genres column:

In [154]:
display_table(google_play_data_final, 9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Below, we can see the most popular apps in Google Play by Category column:

In [155]:
display_table(google_play_data_final, 1)

FAMILY : 18.896660649819495
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

Below, we can see the most popular apps in Apply Store by prime_genre column:

In [156]:
display_table(apple_store_final, 11)

Games : 58.144585789636984
Entertainment : 7.880856345020168
Photo & Video : 4.964318957493019
Education : 3.661185231151101
Social Networking : 3.288861309339125
Shopping : 2.6062674526838348
Utilities : 2.513186472230841
Sports : 2.140862550418864
Music : 2.04778156996587
Health & Fitness : 2.016754576481539
Productivity : 1.7375116351225566
Lifestyle : 1.5823766677008997
News : 1.3341607198262488
Travel : 1.2410797393732547
Finance : 1.1169717654359292
Weather : 0.8687558175612783
Food & Drink : 0.8067018305926157
Reference : 0.5584858827179646
Business : 0.5274588892336333
Book : 0.43437790878063914
Navigation : 0.18616196090598822
Medical : 0.18616196090598822
Catalogs : 0.12410797393732546


### We are analyzing the frequency table generated for the prime_genre column of the App Store dataset.

#### What is the most common genre? What is the next most common?

Games is the most common genre with 58.14%. 

The next one is entertainment with 7.88%.

#### What other patterns do you see?

The majority of the apps are not developed for productive purposes.


#### What is the general impression — are most of the apps designed for practical purposes (education, shopping, utilities, productivity, lifestyle) or more for entertainment (games, photo and video, social networking, sports, music)?

Most of the apps are designed for entertainment.


#### Can you recommend an app profile for the App Store market based on this frequency table alone? 

No, I need more information to make a recommendation.

#### If there's a large number of apps for a particular genre, does that also imply that apps of that genre generally have a large number of users? 

Probably.


### We are analyzing the frequency table generated for the prime_genre column of the Google Play dataset.

#### What are the most common genres?

Family is the most popular with 18.89%.

#### What other patterns do you see?

Game is the second most popular with 9.72%.

#### Can you recommend an app profile based on what you found so far? Do the frequency tables you generated reveal the most frequent app genres or what genres have the most users?

No, I need more information to make a recommendation.


### Data Analysis part #2

#### Apple Store analysis

The code below creates a frequency table of the Apple Store datase

In [157]:
apple_store_freq_table = freq_table(apple_store_final, 11)

The code below creates a frequency table of the Apple Store dataset based on the application genre and the average of user reviews. The task aim is to identify which genre has the most user active user.

In [158]:
apple_store_genre_rating_avg = dict()
for genre in apple_store_freq_table:
    total = 0
    len_genre = 0
    
    for row in apple_store_final[1:]:
        genre_app = row[11]
        if genre_app ==  genre:
            rating_count_tot = float(row[5])
            total += rating_count_tot
            len_genre += 1
    avg_user_rating = total / len_genre    
    apple_store_genre_rating_avg[genre] = round(avg_user_rating)
    

In [159]:
print(apple_store_genre_rating_avg)

{'Social Networking': 71548, 'Photo & Video': 28442, 'Games': 22789, 'Music': 57327, 'Reference': 74942, 'Health & Fitness': 23298, 'Weather': 52280, 'Utilities': 18684, 'Travel': 28244, 'Shopping': 26920, 'News': 21248, 'Navigation': 86090, 'Lifestyle': 16486, 'Entertainment': 14030, 'Food & Drink': 33334, 'Sports': 23009, 'Book': 39758, 'Finance': 31468, 'Education': 7004, 'Productivity': 21028, 'Business': 7491, 'Catalogs': 4004, 'Medical': 612}


Below, we can see the ranking of the application genre based on the average number of user reviews:

In [160]:
sorted(apple_store_genre_rating_avg, key=apple_store_genre_rating_avg.get, reverse=True)

['Navigation',
 'Reference',
 'Social Networking',
 'Music',
 'Weather',
 'Book',
 'Food & Drink',
 'Finance',
 'Photo & Video',
 'Travel',
 'Shopping',
 'Health & Fitness',
 'Sports',
 'Games',
 'News',
 'Productivity',
 'Utilities',
 'Lifestyle',
 'Entertainment',
 'Business',
 'Education',
 'Catalogs',
 'Medical']

#### Analyze the results and try to come up with at least one app profile recommendation for the App Store

My recommendation profile is Navigation. 

This genre category contains the most user reviews with 86,090 user reviews on average.

#### Google Play analysis

In [161]:
google_play_freq_table = freq_table(google_play_data_final, 1)
    

In [162]:
print(google_play_freq_table)

{'ART_AND_DESIGN': 0.6430505415162455, 'AUTO_AND_VEHICLES': 0.9250902527075812, 'BEAUTY': 0.5979241877256317, 'BOOKS_AND_REFERENCE': 2.1435018050541514, 'BUSINESS': 4.591606498194946, 'COMICS': 0.6204873646209386, 'COMMUNICATION': 3.2378158844765346, 'DATING': 1.861462093862816, 'EDUCATION': 1.1620036101083033, 'ENTERTAINMENT': 0.9589350180505415, 'EVENTS': 0.7107400722021661, 'FINANCE': 3.7003610108303246, 'FOOD_AND_DRINK': 1.2409747292418771, 'HEALTH_AND_FITNESS': 3.0798736462093865, 'HOUSE_AND_HOME': 0.8235559566787004, 'LIBRARIES_AND_DEMO': 0.9363718411552346, 'LIFESTYLE': 3.9034296028880866, 'GAME': 9.724729241877256, 'FAMILY': 18.896660649819495, 'MEDICAL': 3.531137184115524, 'SOCIAL': 2.6624548736462095, 'SHOPPING': 2.2450361010830324, 'PHOTOGRAPHY': 2.944494584837545, 'SPORTS': 3.395758122743682, 'TRAVEL_AND_LOCAL': 2.33528880866426, 'TOOLS': 8.461191335740072, 'PERSONALIZATION': 3.3167870036101084, 'PRODUCTIVITY': 3.892148014440433, 'PARENTING': 0.6543321299638989, 'WEATHER': 

In [163]:
google_play_install_avg = dict()
for category in google_play_freq_table:
    total = 0
    len_category = 0
    
    for row in google_play_data_final[1:]:
        category_app = row[1]        
        if category_app == category:
            installs = row[5].replace('+', "")
            installs = float(installs.replace(',', ""))
            total += installs
            len_category += 1
    installs_avg = total / len_category    
    google_play_install_avg[category] = round(installs_avg, 2)
        
                        
        

In [164]:
print(google_play_install_avg)

{'ART_AND_DESIGN': 1986335.09, 'AUTO_AND_VEHICLES': 647317.82, 'BEAUTY': 513151.89, 'BOOKS_AND_REFERENCE': 8767811.89, 'BUSINESS': 1712290.15, 'COMICS': 817657.27, 'COMMUNICATION': 38456119.17, 'DATING': 854028.83, 'EDUCATION': 1833495.15, 'ENTERTAINMENT': 11640705.88, 'EVENTS': 253542.22, 'FINANCE': 1387692.48, 'FOOD_AND_DRINK': 1924897.74, 'HEALTH_AND_FITNESS': 4188821.99, 'HOUSE_AND_HOME': 1331540.56, 'LIBRARIES_AND_DEMO': 638503.73, 'LIFESTYLE': 1437816.27, 'GAME': 15588015.6, 'FAMILY': 3697848.17, 'MEDICAL': 120550.62, 'SOCIAL': 23253652.13, 'SHOPPING': 7036877.31, 'PHOTOGRAPHY': 17840110.4, 'SPORTS': 3638640.14, 'TRAVEL_AND_LOCAL': 13984077.71, 'TOOLS': 10801391.3, 'PERSONALIZATION': 5201482.61, 'PRODUCTIVITY': 16787331.34, 'PARENTING': 542603.62, 'WEATHER': 5074486.2, 'VIDEO_PLAYERS': 24727872.45, 'NEWS_AND_MAGAZINES': 9549178.47, 'MAPS_AND_NAVIGATION': 4056941.77}


In [165]:
sorted(google_play_install_avg, key=google_play_install_avg.get, reverse=True)

['COMMUNICATION',
 'VIDEO_PLAYERS',
 'SOCIAL',
 'PHOTOGRAPHY',
 'PRODUCTIVITY',
 'GAME',
 'TRAVEL_AND_LOCAL',
 'ENTERTAINMENT',
 'TOOLS',
 'NEWS_AND_MAGAZINES',
 'BOOKS_AND_REFERENCE',
 'SHOPPING',
 'PERSONALIZATION',
 'WEATHER',
 'HEALTH_AND_FITNESS',
 'MAPS_AND_NAVIGATION',
 'FAMILY',
 'SPORTS',
 'ART_AND_DESIGN',
 'FOOD_AND_DRINK',
 'EDUCATION',
 'BUSINESS',
 'LIFESTYLE',
 'FINANCE',
 'HOUSE_AND_HOME',
 'DATING',
 'COMICS',
 'AUTO_AND_VEHICLES',
 'LIBRARIES_AND_DEMO',
 'PARENTING',
 'BEAUTY',
 'EVENTS',
 'MEDICAL']

Analyze the results and try to come up with at least one app profile recommendation for the Google Play.

My recommendation profile is Communicaton.

This category contains the most application installation, with 38,456,119.17 cases on average.